asv_bench/benchmarks/strings.py

```diff
@@ -87,6 +84,29 @@ def time_repeat(self, repeats):
         self.s.str.repeat(self.repeat)
 
 
+class Cat(object):
+
+    goal_time = 0.2
+    params = ([None, 5], [None, ','], [None, '-'], [0.0, 1e-4, 0.1])
+    param_names = ['others', 'sep', 'na_rep', 'na_frac']
+
+    def setup(self, others, sep, na_rep, na_frac):
+        N = int(5e5)
+        mask_gen = lambda: np.random.choice([True, False], N,
```

Comment:
Do we need the lambda here? Figured we could just use the returned array in the subsequent call

Reply:
Yes, it's essential IMO that each column gets a different mask to compound the NaNs (for the `na_rep=None` case) and have a reasonable similarity to real-life data.